src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedDeletedMethod.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Reflection;
+using Microsoft.Cci;
+using Microsoft.CodeAnalysis.CodeGen;
+using Microsoft.CodeAnalysis.Symbols;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// Represents a method that has been deleted in an Edit and Continue session, and whose body
+    /// throws a System.MissingMethodException.
+    /// </summary>
+    internal sealed class SynthesizedDeletedMethod : MethodSymbol, ISynthesizedDeletedMethod
+    {
+        private readonly MethodSymbol _methodSymbol;
+        private readonly NamedTypeSymbol _containingTypeSymbol;
+
+        internal SynthesizedDeletedMethod(MethodSymbol methodSymbol, NamedTypeSymbol containingTypeSymbol)
+            : base()
+        {
+            _methodSymbol = methodSymbol;
+            _containingTypeSymbol = containingTypeSymbol;
+        }
+
+        // MethodSymbol properties that delegate through _containingTypeSymbol
+
+        internal override ModuleSymbol ContainingModule => _containingTypeSymbol.ContainingModule;
+
+        public override Symbol ContainingSymbol => _containingTypeSymbol;
+
+        public override AssemblySymbol ContainingAssembly => _containingTypeSymbol.ContainingAssembly;
+
+        public override NamespaceSymbol ContainingNamespace => _containingTypeSymbol.ContainingNamespace;
+
+        public override bool IsImplicitlyDeclared => true;
+
+        internal override bool SynthesizesLoweredBoundBody => true;
+
+        // MethodSymbol properties that delegate through _methodSymbol
+
+        public override string Name => _methodSymbol.Name;
+
+        public override MethodKind MethodKind => _methodSymbol.MethodKind;
+
+        public override int Arity => _methodSymbol.Arity;
+
+        public override bool IsExtensionMethod => _methodSymbol.IsExtensionMethod;
+
+        public override bool HidesBaseMethodsByName => _methodSymbol.HidesBaseMethodsByName;
+
+        public override bool IsVararg => _methodSymbol.IsVararg;
+
+        public override bool ReturnsVoid => _methodSymbol.ReturnsVoid;
+
+        public override bool IsAsync => _methodSymbol.IsAsync;
+
+        public override RefKind RefKind => _methodSymbol.RefKind;
+
+        public override TypeWithAnnotations ReturnTypeWithAnnotations => _methodSymbol.ReturnTypeWithAnnotations;
```

Comment:
>_methodSymbol.ReturnTypeWithAnnotations [](http://example.com/codeflow?start=73&length=39)

If `_methodSymbol` is from another compilation, then the return type, parameter types, type parameters, etc. from `_methodSymbol` might also be from that compilation, which is likely incorrect.